src/List/Extra.elm

```diff
@@ -100,23 +100,15 @@ last items =
 {-| Return all elements of the list except the last one.
 
     init [ 1, 2, 3 ]
-    --> Just [ 1, 2 ]
+    --> [ 1, 2 ]
 
     init []
-    --> Nothing
+    --> []
 
 -}
-init : List a -> Maybe (List a)
+init : List a -> List a
 init items =
-    case items of
-        [] ->
-            Nothing
-
-        nonEmptyList ->
-            nonEmptyList
-                |> List.reverse
-                |> List.tail
-                |> Maybe.map List.reverse
+    List.take (List.length items - 1) items
```

Comment:
Why not keep the signature as it was? The empty check should be pretty quick.

Reply:
Yeah, if the original signature is preferable I can put it back. Honestly I like the `List.take` / `List.drop` kind of API, where it returns a List, more.

Reply:
I think it's better. A breaking change to this package (which so many other depends on) is not going to be fun to see :sweat_smile: